addons/missileguidance/functions/fnc_ahr_onFired.sqf

```diff
@@ -0,0 +1,71 @@
+#include "script_component.hpp"
+/*
+ * Author: Brandon (TCVM)
+ * Sets up SACLOS state arrays (called from missileGuidance's onFired).
+ *
+ * Arguments:
+ * Guidance Arg Array <ARRAY>
+ *
+ * Return Value:
+ * None
+ *
+ * Example:
+ * [] call ace_missileguidance_fnc_SACLOS_onFired
+ *
+ * Public: No
+ */
+params ["_firedEH", "_launchParams", "", "", "_stateParams"];
+_firedEH params ["_shooter","","","","","","_projectile"];
+_stateParams params ["", "_seekerStateParams"];
+_launchParams params ["_target"];
+
+_target = missileTarget _projectile;
+if (isNull _target && isVehicleRadarOn vehicle _shooter) then {
+    _target = cursorTarget;
+};
+if !(_target isKindOf "AllVehicles") then {
+    _target = nil;
+};
+_launchParams set [0, _target];
+_projectile setMissileTarget objNull; // to emulate a no launch warning
+
+private _projectileConfig = [_projectile] call CBA_fnc_getObjectConfig;
+private _config = _projectileConfig >> "ace_missileguidance";
+
+private _isActive = false;
+private _activeRadarDistance = [_config >> "activeRadarEngageDistance", "NUMBER", 500] call CBA_fnc_getConfigEntry;
+private _projectileThrust = [_projectileConfig >> "thrust", "NUMBER", 0] call CBA_fnc_getConfigEntry;
+private _projectileThrustTime = [_projectileConfig >> "thrustTime", "NUMBER", 0] call CBA_fnc_getConfigEntry;
+
+private _velocityAtImpact = _projectileThrust * _projectileThrustTime;
+private _timeToActive = 0;
+if (!isNil "_target" && _velocityAtImpact > 0) then {
+    private _distanceUntilActive = (((getPosASL _shooter) vectorDistance (getPosASL _target)) - _activeRadarDistance);
+    _timeToActive = 0 max (_distanceUntilActive / _velocityAtImpact);
+};
+
+if (isNil "_target") then {
+    _timeToActive = 0;
+    _isActive = true;
+    _target = objNull;
+};
+
+private _shooterHasActiveRadar = {
```

Comment:
seems like data link must be enabled, so we could also check
`vehicleReportRemoteTargets vehicle _shooter` as well

Reply:
I didn't do that because the datalink to a missile is different to the datalink to aircraft in real life so you don't need to enable datalink to guide a missile